flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/ArcticRowDataTaskWriterFactory.java

```diff
@@ -59,12 +59,7 @@ public void setTransactionId(Long transactionId) {
 
   @Override
   public void initialize(int taskId, int attemptId) {
-    if (table.isKeyedTable()) {
-      Preconditions.checkNotNull(transactionId, "TransactionId should be set first. " +
-          "Invoke setTransactionId() before this method");
-    } else {
-      Preconditions.checkArgument(transactionId == null, "TransactionId should be null for unkeyed table.");
-    }
+    Preconditions.checkArgument(transactionId == null, "TransactionId should be null for table.");
```

Comment:
I think the variable transactionId is the `final` variable which is better.
`private transient final Long transactionId = null;`

Reply:
I added `final` to it.